kdumpctl

```diff
@@ -1659,47 +1662,156 @@ _should_reset_crashkernel() {
 	[[ $(kdump_get_conf_val auto_reset_crashkernel) != no ]] && systemctl is-enabled kdump &> /dev/null
 }
 
+set_kdump_test_id()
+{
+	local _id=$1
+	local _kmsg="/dev/kmsg"
+
+	if [[ $DEFAULT_DUMP_MODE == "fadump" ]]; then
+		[[ -c "$_kmsg" ]] && echo "$_id" > "$_kmsg"
+	else
+		KDUMP_COMMANDLINE_APPEND+=" $_id "
+		reload >& /dev/null
+	fi
+
+	if [[ "$?" -ne 0 ]]; then
+		derror "Set kdump test id fail."
+		exit 1
+	fi
+}
+
+# $1: success/fail/pending/manual/clear
+# $2: test id
+set_vmcore_creation_status()
+{
+	local _status=$1
+	local _test_id=$2
+	_dir=$(dirname "$VMCORE_CREATION_STATUS")
+
+	[[ -d "$_dir" ]] || mkdir -p "$_dir"
+	[[ -w "$_dir" ]] || chmod +w "$_dir"
+
+	case "$_status" in
+		pending)
+			echo  "$_status $_test_id" > "$VMCORE_CREATION_STATUS"
+			;;
+		success | fail | manual)
+			sed -E -i "s/^\w+/$_status/" "$VMCORE_CREATION_STATUS"
+			;;
+		clear)
+			rm -f "$VMCORE_CREATION_STATUS"
+			;;
+		*)
+			return
+	esac
+	sync -f "$_dir"
+}
+
+fetch_test_status()
+{
+	local _test_id="$1"
+	local _script="$KDUMP_TMPDIR/fetch.sh"
+
+	cat << EOF > "$_script"
+#!/bin/bash
+dmesg_files=\$(find "\$1" -name "kexec-dmesg.log" -or -name "vmcore-dmesg.txt" 2> /dev/null)
+[[ -z "\$dmesg_files" ]] && exit 2
+dmesg_files=\$(ls -t \$dmesg_files | head -50)
+id_file=\$(grep -l "$_test_id" \$dmesg_files | uniq)
+case \$(echo -n "\$id_file" | wc -w) in
+	0)
+		# Not found files containing the test id
+		exit 2
+		;;
+	*)
+		# Found one or more files containing the test id
+		dir=\$(dirname \$id_file | uniq)
+		case \$(echo -n "\$dir" | wc -w) in
+			1)
+				# The test id files within one folder
+				;;
+			*)
+				# The test id files shouldn't within
+				# multiple folders
+				exit 2
+				;;
+		esac
+		;;
+esac
+if [[ -f "\$dir/vmcore" || -f "\$dir/vmcore.flat" ]]; then
+	# Success
+	exit 0
+else
+	# Fail
+	exit 1
+fi
+EOF
+	if is_nfs_dump_target || is_local_target; then
+		mkdir -p "$TMPMNT"
+		mount "${OPT[_target]}" "$TMPMNT" -t "${OPT[_fstype]}" -o defaults || \
+			{ dwarn "Failed to mount ${OPT[_target]}" && return 2; }
+		bash "$_script" "$TMPMNT/${OPT[path]}"
+	elif is_ssh_dump_target; then
```

Comment:
I believe we should only mount the target when it isn't mounted already.

Reply:
The reason I didn't check the mount is, it seems not a big issue to me if the drive is already mounted before, and we mount the same drive to a different place again. Since a) we are in 1st kernel here, memory usage is not a problem, and b) we only read into the drive, no write operations, so it is harmless, and c) extra code will be involved for mount checking and switching to the mounting, seems redundant and unecessary. What do you think?

Reply:
In v2, I checked if the target is mounted or not before mount it into tmpmnt. So the device will only mount once.